src/containers/layout/navbar/NavBar.jsx

```diff
@@ -22,19 +22,24 @@ import { student, tutor } from '~/constants'
 import { styles } from '~/containers/layout/navbar/NavBar.styles'
 
 const Navbar = () => {
-  const [navigationItems, setNavigationItems] = useState(
-    Object.values(guestRoutes.navBar)
-  )
   const { userRole } = useSelector((state) => state.appMain)
   const { openDrawer, closeDrawer, isOpen } = useDrawer()
   const { t } = useTranslation()
 
-  useEffect(() => {
-    if (userRole === student)
-      setNavigationItems(Object.values(studentRoutes.navBar))
-    else if (userRole === tutor)
-      setNavigationItems(Object.values(tutorRoutes.navBar))
-    else setNavigationItems(Object.values(guestRoutes.navBar))
+  const navigationItems = useMemo(() => {
+    return userRole === student
+      ? Object.values(studentRoutes.navBar)
+      : userRole === tutor
+      ? Object.values(tutorRoutes.navBar)
+      : Object.values(guestRoutes.navBar)
+  }, [userRole])
+
+  const accountItems = useMemo(() => {
+    return userRole === student
```

Comment:
double ternary is bad for readability, also it will cause code smell in sonar

Reply:
fixed